include/exadg/solvers_and_preconditioners/solvers/iterative_solvers_dealii_wrapper.h

```diff
@@ -122,9 +122,12 @@ class SolverCG : public SolverBase<VectorType>
   void
   update_preconditioner(bool const update_preconditioner) const override
   {
-    if(solver_data.use_preconditioner and update_preconditioner)
+    if(solver_data.use_preconditioner)
     {
-      preconditioner.update();
+      if(preconditioner.needs_update() or update_preconditioner)
```

Comment:
ok so actually, `needs_update` is only `false` initially and causes a preconditioner update in the very first solver call. Afterwards, we force the update from the linear solver call, right?

Reply:
Yes, exactly. Let's put it this way: "The user is the king.", i.e. if the user says the preconditioner should be updated, we will do it. "With great power comes great responsibility", though ;)